src/components/esn-app-grid/esn-app-grid.tsx

```diff
@@ -0,0 +1,103 @@
+import { Component, Prop, ComponentInterface, State, Host, h } from '@stencil/core';
+import { Application } from './esn-app-grid.types';
+import { AppGridTogglerIcon } from './icons/AppGridIcon';
+import { CalendarAppIcon } from './icons/CalendarAppIcon';
+import { InboxAppIcon } from './icons/InboxAppIcon';
+import { AdminAppIcon } from './icons/AdminAppIcon';
+import { ContactsAppIcon } from './icons/ContactsAppIcon';
+import { LinShareAppIcon } from './icons/LinShareAppIcon';
+import { UnknownAppIcon } from './icons/UnknownAppIcon';
+
+const iconMapping = {
+  Admin: AdminAppIcon,
+  Calendar: CalendarAppIcon,
+  Contacts: ContactsAppIcon,
+  Inbox: InboxAppIcon,
+  LinShare: LinShareAppIcon
+};
+
+const isIconAvailable = (appName: string) => Object.keys(iconMapping).includes(appName);
+
+@Component({
+  tag: 'esn-app-grid',
+  styleUrl: 'esn-app-grid.scss',
+  shadow: true
+})
+export class EsnAppGrid implements ComponentInterface {
+  private popover: HTMLElement;
+
+  @State() applications: Application[] = [];
+
+  /**
+   * A serialized JSON object that contains information about the available apps.
+   */
+  @Prop() serializedApplications: string;
+
+  componentWillLoad() {
+    try {
+      if (!this.serializedApplications) {
+        throw new Error("The 'serializedApplications' attribute/property must be set to a serialized JSON");
+      }
+
+      const applications = JSON.parse(this.serializedApplications);
+
+      if (Array.isArray(applications) && applications.length) {
+        this.applications = applications.map(({ name, url }) => {
+          if (!isIconAvailable(name)) {
+            console.warn(
+              `The application with name ${name} does not have an icon available. A fallback icon will be used instead.`
+            );
+          }
+
+          return {
+            name,
+            url
+          };
+        });
+      } else {
+        throw new Error("The parsed 'serializedApplications' has an invalid JSON structure");
+      }
```

Comment:
We used to do the other way:
```js
if (!Array.isArray(applications) || !applications.length) {
  throw new Error();
}

// continue your work
```

Reply:
Yeah this is the way we've been used to doing this whole time. Fixed.